### PR TITLE
Upgrade to Rails 5.1

### PR DESCRIPTION
**Pull request by sirwolfgang**

### CODE REVIEW

File: config/boot.rb [hunk 1]
```diff
@@ -1,13 +1,3 @@
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
 
 require 'bundler/setup' # Set up gems listed in the Gemfile.
-
```
Comment (sdomino) [inline config/boot.rb:4]: Is there any reason not to leave this in and just update it to work with Rails 5? My thought is, this way you don't have to always remember to start your server with `-b 0.0.0.0 -p 8080` and wonder why it's not working if you forget to do that. Thoughts?

```
require 'rails/command
require 'rails/commands/server/server_command'
module Rails		
  class Server		
    alias :_default_options :default_options		
    def default_options		
      _default_options.merge!(Host:'0.0.0.0', Port:8080)		
    end		
  end		
end
```

File: boxfile.yml [hunk 2]
```diff
@@ -22,7 +26,7 @@ data.db:
 web.main:
   start:
     nginx: nginx -c /app/config/nginx.conf
-    puma: bundle exec puma -C /app/config/puma.rb
+    puma: bundle exec rails s -b 0.0.0.0 -p 8080
```
Comment (sdomino) [inline boxfile.yml:29]: Any particular reason you pulled out puma here?

**Review by sdomino — COMMENTED** [review]
This is great, thanks! You'll find a few comments I'd like your feedback on, otherwise, I think it's ready to merge.

Also, it looks like it's missing the `rake db:setup_or_migrate` task. Can we either get that added, or just change the boxfile to `rake db:migrate`. That way it should work in production (`nanobox deploy dry-run`)

Something like this:
```
desc 'either sets up the db or migrates it depending on state of db'
task setup_or_migrate: :environment do
  begin
    ActiveRecord::Base.connection
  rescue ActiveRecord::NoDatabaseError
    Rake::Task["db:setup"].invoke
  else
    Rake::Task["db:migrate"].invoke
  end
end
```

File: boxfile.yml (hunk 2 above)
Comment (sirwolfgang) [reply boxfile.yml:29]: It still boots and uses puma, as intended, I believe this to be the more conical way to boot the rails app. 

Part of this was to pass in the binding and port settings, which were set before via a monkey patching hack that is no longer supported with Rails 5.1

File: config/boot.rb (hunk 1 above)
Comment (sirwolfgang) [reply config/boot.rb:4]: This is no longer supported via Rails 5.1. The internal structure changed, and I did not see a way to reimplement this. I would also suggest this method of monkey patching is less than ideal and prone to breaking in such cases like an update or patch of rails.

**Pull request merged**